### DISCUSSION TITLE
Displaying a flag based on value based on multiple columns

### DISCUSSION THREAD

**Discussion by exorgroup** [Laravel-Backpack/community-forum#735] [Q&A (Help)] [answered]
In a list I would like to display a svg image showing if the record has been tempered with or not.  For this reason I have a hash value stored in the database.  I need to compare the hash value in the database with the one generated for the data.  If they match than I want to display a particular svg image, and if they do not match I want to display another svg image.

My idea was to use a custom column but have stumbled upon a block - one table my have 3 columns, another may have 10 columns whilst another may have 7 columns and do not want to create a custom column for each screen that I have to perform the same function

Could do with placing everything in an array and than go through the array in the function but do not know how I can get the data (all columns values) in an array and passed to the function evaluated there and the right svg image displayed

Any suggestion is welcome

**Comment by tabacitu** [comment]
Hmm... I would probably use the `text` column for that, but instead of showing the actual text I'd pass a closure:

```php
CRUD::column('name')->value(function($entry) {
    if ($entry->isTampered('name') {
        return  '<img src="red.png">' . $name;
    }

    return '<img src="green.png">' . $name;
})->escaped(false);
```

Now that I write it though... I don't particularly love it. Another option would be to create ONE custom column (`text_with_tampered_icon`) starting from the `text` column, that also includes the logic above, an calls `isTampered($columnName)` on the Model. That way I'd end up with:

```php
CRUD::column('name')->type('text_with_tampered_icon');
```

So then my actual answer is... I'd do that on the model 😀 I would have a function on the model that just returns true/false if a particular column was tampered with, and use that in my admin panel / wherever else.

Hope it helps.

**Reply by siberfx** [reply]
but when it's a closure, in search input it doesnt work. then you have to add searchLogic as false.

**Reply by exorgroup** [reply]
The example seen above by @tabacitu will only work with 1 column such as name or id - unfortunately will not work for what I need

To give a better example consider the table has 4 columns of user data such as id, name, surname and age.  A 5th column stores my hash value of id, name, surname and age 

In the function I need to pass the id, name, surname and age (4 fields total) than calculate the new hash and compare that result with the hash in the database.

Is there a way that I can pass the entire row back to the model and then iterate through its values?

**Reply by tabacitu** [reply]
Hmmm I'm afraid I don't understand. 

> To give a better example consider the table has 4 columns of user data such as id, name, surname and age. A 5th column stores my hash value of id, name, surname and age

Even with your example, the way I see it: the `$entry` that you have in the closure is the Model entry; the database row; so it has all db columns. So on your `$entry` you can do:
- `$entry->id`
- `$entry->name`
- `$entry->surname`
- `$entry->age`
- `$entry->hashes`

And even in a model function, you can work with all of those, because you're on the Model level. For example:

```php
class User extends Model {
    public function isTampered($column) {
        // use $this->{$column} to get that column
        // use $this->hashes to get the hashes
        // compare them or whatever
        return true; // or false
    }
}
```

> Is there a way that I can pass the entire row back to the model and then iterate through its values?

The row IS the model row. You don't have to pass it. `$entry` is the model row. So in the example above, it will be a User model.

**Reply by exorgroup** [reply, marked as answer]
I got what I need and I am sharing so that I learn if it can be improved 

this is my column definition
```php
CRUD::column([
            // run a function on the CRUD model and show its return value
            'name'  => 'Hash',
            'label' => 'Hash', // Table column heading
            'type'  => 'model_function',
            'escaped' => false,
            'function_name' => 'getHash', // the method in your Model
             'limit' => 2000, // Limit the number of characters shown
         ]);
``` 


in model 
```php
 public function getHash() {
        if (Security::validData(array($this->name, $this->surname,$this->age), $this->HASH)) {
            return  '<img src="green.png">';
        }

       return '<img src="red.png">';
    }
```

the attributes can be found in model at
```php
protected $fillable = [
        'name',
        'surname',
        'age',
        'HASH'
    ];
```

HASH is a value in my database 
Security::validData is my function to generate hash and evaluate - will not share details of but can let you know it returns TRUE or FALSE

**Reply by tabacitu** [reply]
Glad you figured it out @exorgroup . Have a good one!
